src/dataset/nuscenes_dataset.py

```diff
@@ -1,26 +1,173 @@
+                                                        inverse=False)
+        trans_matrix = np.dot(car_to_global, sensor_to_car)
+        return cal_sensor_data, pose_data, trans_matrix
+
+    def _get_calibration_matrix(self, camera_intrinsic):
+        calibration = np.eye(4, dtype=np.float32)
+        calibration[:3, :3] = camera_intrinsic
+        calibration = calibration[:3]
+        return calibration
+
+    def _get_annotations(self, boxes, calibration, camera_intrinsic, trans_matrix):
+        annotations = []
+        for box in boxes:
+            det_name = nuscenes.eval.detection.utils.category_to_detection_name(box.name)
+            if det_name is None:
+                continue
+
+            ann = self._get_annotation(box, calibration, camera_intrinsic, det_name, trans_matrix)
+            annotations.append(ann)
+
+        annotations = self._get_visible_annotations(annotations)
+        return annotations
+
+    def _get_annotation(self, box, calibration, camera_intrinsic, det_name, trans_matrix):
+        v = np.dot(box.rotation_matrix, np.array([1, 0, 0]))
+        yaw = -np.arctan2(v[2], v[0])
+        box.translate(np.array([0, box.wlh[2] / 2, 0]))
+        category_id = self.cat_ids[det_name]
+        amodel_center = dataset_utils.project_to_image(
+                np.array([box.center[0], box.center[1] - box.wlh[2] / 2, box.center[2]], np.float32).reshape(1, 3),
+                calibration)[0].tolist()
+        sample_ann = self.nusc.get('sample_annotation', box.token)
+        instance_token = sample_ann['instance_token']
+        vel = self.nusc.box_velocity(box.token)
+        vel = np.dot(np.linalg.inv(trans_matrix), np.array([vel[0], vel[1], vel[2], 0], np.float32)).tolist()
+        ann = {
+            'category_id':   category_id,
+            'dim':           [box.wlh[2], box.wlh[0], box.wlh[1]],
+            'location':      [box.center[0], box.center[1], box.center[2]],
+            'depth':         box.center[2],
+            'occluded':      0,
+            'truncated':     0,
+            'rotation_y':    yaw,
+            'amodel_center': amodel_center,
+            'iscrowd':       0,
+            'track_id':      instance_token,
+            'velocity':      vel
+        }
+        bbox = kitti.KittiDB.project_kitti_box_to_image(copy.deepcopy(box), camera_intrinsic, imsize=(1600, 900))
+        alpha = dataset_utils.rot_y2alpha(yaw, (bbox[0] + bbox[2]) / 2, camera_intrinsic[0, 2], camera_intrinsic[0, 0])
+        ann['bbox'] = [bbox[0], bbox[1], bbox[2] - bbox[0],
+                       bbox[3] - bbox[1]]
+        ann['area'] = (bbox[2] - bbox[0]) * (bbox[3] - bbox[1])
+        ann['alpha'] = alpha
+        return ann
+
+    def _get_visible_annotations(self, anns):
+        visible_anns = []
+        for i in range(len(anns)):
+            vis = True
+            for j in range(len(anns)):
+                if anns[i]['depth'] - min(anns[i]['dim']) / 2 > anns[j]['depth'] + max(
```

Comment:
This line here is a bit complex. Why not save the two comparison results as variables and then just evaluate them in the `if` block? By saving the comparisons in a named variable, you can also be more descriptive of what each operation does.